### ISSUE TITLE
Incorrectly parses nested statements

### ISSUE THREAD

**Issue by typio** [toji/wgsl-preprocessor#2]
I believe I've found a failing test case
```ts
expect(wgsl`
  #if ${true}
      Yes
  #elif ${false}
      No1
      #if ${false}
          No2
      #else
          No3
      #endif
      No4
      #if ${false}
          No5
      #else
          No6
      #endif
  #endif`.replace(/\s/g,'')).to.equal(`Yes`);
```
this outputs `'YesNo4No6'` instead of `'Yes'`, it appears the nested `#endif` is getting applied to the outer `#if`.

I can try to fix this if you'd like.

**Comment by fynv** [comment]
I have repro of the bug in my project too.

**Comment by fynv** [comment]
I find that there's a simple alternative:

```js
function condition(cond, a, b="")
{
    return cond? a: b;
}

let code = `

${condition(condA,`
    some_code_A();
`)}

${condition(condB,`
    some_code_B1();
`,`
    some_code_B2();
`)}

`;


```

**Comment by toji** [comment]
Thank you for the bug report, and sorry it took me 9 months to get around to fixing it!

I've made your test case part of the official test suite, updated the library to fix it (and generally be less fragile all around) and published it to npm and v1.0.1. Thanks!
